utoipa-gen/src/component.rs

```diff
@@ -1306,16 +1307,18 @@ impl ComponentSchema {
 
                         schema.to_tokens(tokens);
                     } else {
+                        // not inline or unset
                         let schema_type = SchemaType {
                             path: Cow::Borrowed(&rewritten_path),
                             nullable,
                         };
-                        let index =
-                            if !schema_type.is_primitive() || type_tree.generic_type.is_none() {
-                                container.generics.get_generic_type_param_index(type_tree)
-                            } else {
-                                None
-                            };
+                        let index = if !is_not_inline
```

Comment:
Here is the line which I in particular wanted to get rid of.
The double negation which is not the same as the positive is quite confusing. Inlining would help here.